### PR TITLE
Entry for contains-group-kinds annotation and updated contains-group-resources annotation

### LINKED ISSUE #47092: kubectl ApplySets require `contains-group-kinds` annotation but documentation references `contains-group-resources`
### What happened?

I'm not entirely sure if this is a bug, or if the documentation is just inaccurate/lacking some context, but it's at the very least a bit confusing.

If you check the documentation for the well-known labels, etc., you can find the documentation for the `applyset.kubernetes.io/contains-group-resources` annotation [here](https://kubernetes.io/docs/reference/labels-annotations-taints/#applyset-kubernetes-io-contains-group-resources).

image

However, when I try to reference a parent object with the `--applyset=` option to `kubectl apply` without this annotation, it complains about the `contains-group-kinds` annotation missing:

```
$ kubectl apply -f - --server-side --applyset applyset/my-applyset --prune
error: parsing ApplySet annotation on "applysets.nyl.io/my-applyset": kubectl requires the "applyset.kubernetes.io/contains-group-kinds" annotation to be set on all ApplySet parent objects
```

At this point my parent resource is declared as:

```yaml
apiVersion: nyl.io/v1
kind: ApplySet
metadata:
  annotations:
    applyset.kubernetes.io/tooling: kubectl/v1.30.1
  labels:
    applyset.kubernetes.io/id: applyset-dppUASvWbAC2LF4ATdrwEpk7WJRt6vIRUnqX_1JY2+s-v1
  name: my-applyset
```

It appears that I can set either one of the two (`*-kinds` or `*-resources`) on the `ApplySet` resource and `kubectl` will accept it; only in the latter case it appears to be copying the "resources" annotation to the add the "kinds" annotation.

To phrase it another way: Setting the "kinds" annotation seems like the right way to go (but not what the documentation suggests), and setting the "resources" annotation works as well but you'll _also_ end up with the "kinds" annotation afterwards.

### What did you expect to happen?

I expect the documentation and kubectl to align.

### How can we reproduce it (as minimally and precisely as possible)?

Point `kubectl apply --applyset=...` at a resource that doesn't have the required annotations.

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.30.1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.1+k3s1
```

</details>

### Cloud provider

nocloud

### OS version

_No response_

### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

_No response_

### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by T-Lakshmi**
Resolves #47092.

Code change done in https://github.com/kubernetes/kubernetes/pull/118942.

### CODE REVIEW

**Comment by T-Lakshmi** [comment]
[Preview page](https://deploy-preview-47120--kubernetes-io-main-staging.netlify.app/docs/reference/labels-annotations-taints/#applyset-kubernetes-io-contains-group-kinds)

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -176,6 +176,31 @@ or use a different optimization. However, as of Kubernetes version {{< skew curr
...
+
+For Kubernetes version {{< skew currentVersion >}}, you can use this annotation on Secrets, ConfigMaps,
+or custom resources if the CustomResourceDefinition
+defining them has the `applyset.kubernetes.io/is-parent-type` label.
+
+Part of the specification used to implement
+[ApplySet-based pruning in kubectl](/docs/tasks/manage-kubernetes-objects/declarative-config/#alternative-kubectl-apply-f-directory-prune).
+This annotation is applied to the parent object used to track an ApplySet to optimize listing of
+ApplySet member objects. It is optional in the ApplySet specification, as tools can perform discovery
+or use a different optimization. However, as of Kubernetes version {{< skew currentVersion >}},
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:195]:
```suggestion
or use a different optimization. However, in Kubernetes version {{< skew currentVersion >}},
```

**Comment by T-Lakshmi** [comment]
/assign @dipesh-rawat,

Hi @dipesh-rawat,
Pinging for PR approval, can you PTAL!

**Review by dipesh-rawat — COMMENTED** [review]
/approve

**Pull request merged**